src/Microsoft.DotNet.Wpf/src/PresentationBuildTasks/Microsoft/Build/Tasks/Windows/GenerateTemporaryTargetAssembly.cs

```diff
@@ -761,52 +762,96 @@ static private void ReplaceImplicitImports(XmlDocument xmlProjectDoc)
             }
 
             XmlNode root = xmlProjectDoc.DocumentElement;
-          
+
             for (int i = 0; i < root.Attributes.Count; i++)
             {
                 XmlAttribute xmlAttribute = root.Attributes[i] as XmlAttribute;
 
                 if (xmlAttribute.Name.Equals("Sdk", StringComparison.OrdinalIgnoreCase))
                 {
-                    // <Project Sdk="Microsoft.NET.Sdk">
+                    //  <Project Sdk="Microsoft.NET.Sdk">
+                    //  <Project Sdk="My.Custom.Sdk/1.0.0" />
+                    //  <Project Sdk="My.Custom.Sdk/min=1.0.0" />
```

Comment:
I didn't know about this API. Should we call `SdkReference.TryParse` directly instead of reproducing the parse here ? It's a public API and it does not require another dependency.

Reply:
If returning early due to a parsing failure, make sure to not remove the element.  

```cs
   static private void ReplaceImplicitImports(XmlDocument xmlProjectDoc)
        {
            if (xmlProjectDoc == null)
            {
                // When the parameters are not valid, simply return it, instead of throwing exceptions.
                return;
            }

            XmlNode root = xmlProjectDoc.DocumentElement;

            for (int i = 0; i < root.Attributes.Count; i++)
            {
                XmlAttribute xmlAttribute = root.Attributes[i] as XmlAttribute;

                if (xmlAttribute.Name.Equals("Sdk", StringComparison.OrdinalIgnoreCase))
                {
                    //  <Project Sdk="Microsoft.NET.Sdk">
                    //  <Project Sdk="My.Custom.Sdk/1.0.0" />
                    //  <Project Sdk="My.Custom.Sdk/min=1.0.0" />

                    // Remove Sdk attribute
                    var sdkValue = xmlAttribute.Value;

                    SdkReference sdkReference;
                    SdkReference.TryParse(sdkValue, out sdkReference);

                    if (sdkReference == null) return;

                    root.Attributes.Remove(xmlAttribute);

                    //
                    // Add explicit top import
                    //
                    //  <Import Project="Sdk.props" Sdk="Microsoft.NET.Sdk" />
                    //
                    XmlNode nodeImportProps = CreateImportProjectSdkNode(xmlProjectDoc, "Sdk.props", sdkReference);

                    // Prepend this Import to the root of the XML document
                    root.PrependChild(nodeImportProps);

                    //
                    // Add explicit bottom import
                    //
                    //  <Import Project="Sdk.targets" Sdk="Microsoft.NET.Sdk" 
                    //
                    XmlNode nodeImportTargets = CreateImportProjectSdkNode(xmlProjectDoc, "Sdk.targets", sdkReference);

                    // Append this Import to the end of the XML document
                    root.AppendChild(nodeImportTargets);
                }
            }
        }

        // Creates an XmlNode that contains an Import Project element
        //
        //  <Import Project="Sdk.props" Sdk="Microsoft.NET.Sdk" />
        static XmlNode CreateImportProjectSdkNode(XmlDocument xmlProjectDoc, string projectAttributeValue, SdkReference sdkReference)
        {
            XmlNode nodeImport = xmlProjectDoc.CreateElement("Import", xmlProjectDoc.DocumentElement.NamespaceURI);
            XmlAttribute projectAttribute = xmlProjectDoc.CreateAttribute("Project");
            projectAttribute.Value = projectAttributeValue;
            XmlAttribute sdkAttributeProps = xmlProjectDoc.CreateAttribute("Sdk");
            sdkAttributeProps.Value = sdkReference.Name;
            nodeImport.Attributes.Append(projectAttribute);
            nodeImport.Attributes.Append(sdkAttributeProps);

            if (!string.IsNullOrEmpty(sdkReference.Version))
            {
                XmlAttribute sdkVersionAttributeProps = xmlProjectDoc.CreateAttribute("Version");
                sdkVersionAttributeProps.Value = sdkReference.Version;
                nodeImport.Attributes.Append(sdkVersionAttributeProps);
            }

            if (!string.IsNullOrEmpty(sdkReference.MinimumVersion))
            {
                XmlAttribute sdkVersionAttributeProps = xmlProjectDoc.CreateAttribute("MinimumVersion");
                sdkVersionAttributeProps.Value = sdkReference.MinimumVersion;
                nodeImport.Attributes.Append(sdkVersionAttributeProps);
            }

            return nodeImport;
        }
    }
```

Reply:
Done in c164066c329e45f7e8909c1adac3d1c32aaa382d. I also modified some comments in the code and added other exemples of explicit imports after the transformation ([Here ](https://github.com/dotnet/wpf/pull/5426/commits/c164066c329e45f7e8909c1adac3d1c32aaa382d#diff-18cd159183499afb9f8652c07b116234f77e50469adf92fc1c57f3c41f029627R787-R788)and [Here](https://github.com/dotnet/wpf/pull/5426/commits/c164066c329e45f7e8909c1adac3d1c32aaa382d#diff-18cd159183499afb9f8652c07b116234f77e50469adf92fc1c57f3c41f029627R799-R800)).